agent/config.go

```diff
@@ -1183,6 +1179,52 @@ func DecodeConfig(r io.Reader) (*Config, error) {
 			"is no longer used. Please remove it from your configuration.")
 	}
 
+	if !reflect.DeepEqual(result.DeprecatedRetryJoinEC2, RetryJoinEC2{}) {
+		m := discover.Config{
+			"provider":          "aws",
+			"region":            result.DeprecatedRetryJoinEC2.Region,
+			"tag_key":           result.DeprecatedRetryJoinEC2.TagKey,
+			"tag_value":         result.DeprecatedRetryJoinEC2.TagValue,
+			"access_key_id":     result.DeprecatedRetryJoinEC2.AccessKeyID,
+			"secret_access_key": result.DeprecatedRetryJoinEC2.SecretAccessKey,
+		}
+		result.RetryJoin = append(result.RetryJoin, m.String())
+		result.DeprecatedRetryJoinEC2 = RetryJoinEC2{}
+
+		fmt.Fprintf(os.Stderr, "==> DEPRECATION: retry_join_ec2 is deprecated."+
+			"Please add %q to retry_join\n", m)
```

Comment:
I'll add the `json:"-"` and redact the output. Since `m` is local to the `if` block I can just overwrite the fields before I print the error message.